tests/spec/README.md

```diff
@@ -0,0 +1,112 @@
+# Spec tests
+
+These tests specify the expected behaviour of ehrQL, and by extension, the QM.
+
+They are also used to generate documentation.
+
+
+## Table of Contents
+
+`toc.py` defines the Table of Contents for the documentation generated by tests. A new spec test
+subdirectory or file must be defined here in order for it to appear in the docs.
+
+## Folder structure
+
+Within the spec directory, each test directory is considered a chapter in the docs, each file
+is considered to be a section, and each test is a paragraph.
+
+e.g.
+```
+tests
+    ├── spec
+    │   ├── aggregate_frame                       <-- chapter
+    │   │   ├── __init__.py                       <-- contains chapter title
+    │   │   ├── test_count_for_patient.py         <-- section
+    │   │   └── test_exists_for_patient.py        <-- section
+```
+
+### Titles
+Chapter titles are specified with a `title` attribute the test directory's `__init__.py`.
+Section titles are specified with a `title` attribute in each test file.
+Paragraph titles are extracted from the test names.
+
+E.g. `test_count_for_patient.py` contains 2 tests:
+- `test_count_for_patient_on_event_frame()`
+- `test_count_for_patient_on_patient_frame()`
+
+Assuming this is included in `toc.py`, the following structure will be generated in the docs:
+
+1. Aggregating event and patient frames  # from `title` in `aggregate_frame/__init__.py`
+1.1 Counting the rows for each patient  # from `title` in `aggregate_frame/test_count_for_patient.py`
+1.1.1 Count for patient on event frame  # from test name
+1.1.2 Count for patient on patient frame  # from test name
+
+
+## Test structure
+
+Spec tests follow a standardised structure in order to allow documentation generation.  Each
+spec test uses the `spec_test` fixture, which has the following components:
+
+```
+spec_test(
+    table_data,                          # defines the tables in the test database
```

Comment:
thought: I read this and got momentarily disorientated by not knowing what `table_data` was.

It might be helpful to say "dictionary that defines the tables in the test database": you can't immediately see this is a dictionary here.